### ISSUE TITLE
systemd-logind.service: Unexpected error response from GetNameOwner(): Connection terminated

### ISSUE THREAD

**Issue by eoli3n** [systemd/systemd#17520]
**systemd version the issue has been seen with**
 > systemd 245 (245.4-4ubuntu3.2)
+PAM +AUDIT +SELINUX +IMA +APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD +IDN2 -IDN +PCRE2 default-hierarchy=hybrid

**Used distribution**
 > Ubuntu 20.04

**Linux kernel version used** (`uname -a`)
 > 5.4.0-52-generic #57-Ubuntu

**CPU architecture issue was seen on**
 > amd64

**Expected behaviour you didn't see**
 > Session to not kick out users

**Unexpected behaviour you saw**
 > Users kicked out

**Steps to reproduce the problem**
 > Hard to tell

On [x2go servers](https://wiki.x2go.org/doku.php), cgroups are used by user to restrict CPU and RAM usage by slice.
I use ``libpam-script`` to run this script
```
#!/bin/bash
PAM_UID=$(id -u "$PAM_USER")
if [ "$PAM_UID" -ge 1000 ]
then
    /bin/systemctl set-property "user-$PAM_UID.slice" CPUWeight=100 MemoryMax=1024M MemorySwapMax=1024M
fi
```
In log of each servers i can see at the exact same time the same set of errors

```
$ journalctl -u systemd-logind -b
nov. 03 15:31:24 prodpeda-x2go-focal4 systemd[1]: ^[[0;1;39msystemd-logind.service: Unexpected error response from GetNameOwner(): Connection terminated
nov. 03 15:31:39 prodpeda-x2go-focal4 systemd[1]: systemd-logind.service: Succeeded.
nov. 03 15:31:39 prodpeda-x2go-focal4 systemd[1]: ^[[0;1;39m^[[0;1;39msystemd-logind.service: Consumed 1.802s CPU time.
nov. 03 15:31:39 prodpeda-x2go-focal4 systemd[1]: systemd-logind.service: Scheduled restart job, restart counter is at 1.
nov. 03 15:31:39 prodpeda-x2go-focal4 systemd[1]: Stopped Login Service.
nov. 03 15:31:39 prodpeda-x2go-focal4 systemd[1]: ^[[0;1;39m^[[0;1;39msystemd-logind.service: Consumed 1.802s CPU time.
nov. 03 15:31:39 prodpeda-x2go-focal4 systemd[1]: Starting Login Service...
nov. 03 15:31:39 prodpeda-x2go-focal4 systemd-logind[626002]: New seat seat0.
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mFailed to add user by file name 21700605, ignoring: Invalid argument
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mFailed to add user by file name 22002539, ignoring: Invalid argument
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mFailed to add user by file name 0, ignoring: Invalid argument
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mFailed to add user by file name 21812612, ignoring: Invalid argument
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mFailed to add user by file name 21503400, ignoring: Invalid argument
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mFailed to add user by file name 21913909, ignoring: Invalid argument
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mFailed to add user by file name 22001931, ignoring: Invalid argument
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mFailed to add user by file name 22007933, ignoring: Invalid argument
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mFailed to add user by file name 21810046, ignoring: Invalid argument
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mFailed to add user by file name 22002563, ignoring: Invalid argument
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mFailed to add user by file name 21911375, ignoring: Invalid argument
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mFailed to add user by file name 21904957, ignoring: Invalid argument
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mFailed to add user by file name 21604087, ignoring: Invalid argument
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mFailed to add user by file name 21904535, ignoring: Invalid argument
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mFailed to add user by file name 22008848, ignoring: Invalid argument
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mFailed to add user by file name 21914191, ignoring: Invalid argument
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mFailed to add user by file name 22013215, ignoring: Invalid argument
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mFailed to add user by file name 21607534, ignoring: Invalid argument
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mFailed to add user by file name 22005695, ignoring: Invalid argument
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mFailed to add user by file name 21707893, ignoring: Invalid argument
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mFailed to add user by file name 21701109, ignoring: Invalid argument
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mFailed to add user by file name 22009051, ignoring: Invalid argument
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mFailed to add user by file name 22017353, ignoring: Invalid argument
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mFailed to add user by file name 21802976, ignoring: Invalid argument
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mFailed to add user by file name 21916547, ignoring: Invalid argument
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mFailed to add user by file name 20141285, ignoring: Invalid argument
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mFailed to add user by file name 22017936, ignoring: Invalid argument
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mFailed to add user by file name 21804063, ignoring: Invalid argument
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mFailed to add user by file name 21714824, ignoring: Invalid argument
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mFailed to add user by file name 22009047, ignoring: Invalid argument
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mFailed to add user by file name 21808696, ignoring: Invalid argument
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mFailed to add user by file name 21808164, ignoring: Invalid argument
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mFailed to add user by file name 21214558, ignoring: Invalid argument
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mFailed to add user by file name 21807676, ignoring: Invalid argument
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mFailed to add user by file name 21719405, ignoring: Invalid argument
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mFailed to add user by file name 22012189, ignoring: Invalid argument
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mFailed to add user by file name 22013352, ignoring: Invalid argument
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mFailed to add user by file name 21901013, ignoring: Invalid argument
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mFailed to add user by file name 124, ignoring: Invalid argument
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mFailed to add user by file name 22003812, ignoring: Invalid argument
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mFailed to add user by file name 22015967, ignoring: Invalid argument
nov. 03 15:31:41 prodpeda-x2go-focal4 systemd-logind[626002]: ^[[0;1;38;5;185m^[[0;1;39m^[[0;1;38;5;185mUser enumeration failed: Invalid argument
```
I think that the first issue: https://github.com/systemd/systemd/issues/15316
triggers the second one : https://github.com/systemd/systemd/issues/16146

Any help on this please ?

**Comment by eoli3n** [comment]
I found that x2go [force to enable lingering](https://github.com/eugenesan/x2goserver/blob/master/x2goserver/bin/x2gostartagent#L545) and that a lot of process remains after user logouts.

I workaround this by:
- disabling the line ``loginctl enable-linger`` in ``/usr/bin/x2gostartagent``, and removing files in ``/var/lib/systemd/linger/``.
- adding ``KillUserProcesses=yes`` in ``/etc/systemd/logind.conf``.

**Comment by poettering** [comment]
First of all, your logind is lacking this fix: ac4e03d45bcf4ad2e570cabdb218e9bac003cc80

Please backport that first.

Then: Your dbus daemon seems to kick logind off the bus. Figure out why that happens. Turn on debug logging in dbus daemon if needed, and figure out why it comes to the decision to kick off logind.

**Comment by poettering** [comment]
Closing due to lack of response.
